From f23b9148a34c13010882a50a009864327295876c Mon Sep 17 00:00:00 2001
From: palinek <palinek@users.noreply.github.com>
Date: Sun, 18 Dec 2016 23:54:07 +0100
Subject: [PATCH 4/4] MathItem: Honor system locale (#52)

---
 providers.cpp | 23 ++++++++++++++++++++++-
 providers.h   |  8 +++-----
 2 files changed, 25 insertions(+), 6 deletions(-)

diff --git a/providers.cpp b/providers.cpp
index 896a33f..3daad5b 100644
--- a/providers.cpp
+++ b/providers.cpp
@@ -791,12 +791,33 @@ bool VirtualBoxProvider::isOutDated() const
 #ifdef MATH_ENABLED
 #include <muParser.h>
 
+class MathItem::Parser : public mu::Parser
+{
+public:
+    static void initLocale()
+    {
+        try
+        {
+            // use the system's locale instead of the "C"
+            s_locale = std::locale{""};
+        } catch (const std::runtime_error & e)
+        {
+            qWarning().noquote() << "Unable to set locale for Math, " << e.what();
+        }
+    }
+};
+static void muParserInitLocale()
+{
+    MathItem::Parser::initLocale();
+}
+Q_COREAPP_STARTUP_FUNCTION(muParserInitLocale);
+
 /************************************************
 
  ************************************************/
 MathItem::MathItem():
         CommandProviderItem(),
-        mParser{new mu::Parser}
+        mParser{new Parser}
 {
     mToolTip =QObject::tr("Mathematics");
     mIcon = XdgIcon::fromTheme("accessories-calculator");
diff --git a/providers.h b/providers.h
index 2234cfe..d62f47b 100644
--- a/providers.h
+++ b/providers.h
@@ -246,16 +246,14 @@ private:
 
 
 #ifdef MATH_ENABLED
-namespace mu
-{
-    class Parser;
-}
 /************************************************
  * Mathematics
  ************************************************/
 class MathItem: public CommandProviderItem
 {
 public:
+    class Parser;
+public:
     MathItem();
     ~MathItem();
 
@@ -263,7 +261,7 @@ public:
     bool compare(const QRegExp &regExp) const;
     virtual unsigned int rank(const QString &pattern) const;
 private:
-    QScopedPointer<mu::Parser> mParser;
+    QScopedPointer<Parser> mParser;
     mutable QString mCachedInput;
 };
 
-- 
2.8.3

